### PR TITLE
[stable/prometheus-adapter] Fixing warning when overriding rules.custom

### LINKED ISSUE #7585: [stable/prometheus-adapter] default rules.custom value incorrect type, leads to warning when overriding
This is the line in question:

https://github.com/helm/charts/blob/c84205c5ff7291cfa6f3d8c12cefcce9bc0ede37/stable/prometheus-adapter/values.yaml#L43

The value should be an array (`[]`).

This results in a warning:

```
$ helm upgrade --install p stable/prometheus-adapter --set rules.default=false --set rules.custom[0]=foo  --dry-run --debug | grep warning
2018/09/06 14:44:12 warning: cannot overwrite table with non table for custom (map[])
```

PR incoming...

### PR DESCRIPTION
**Pull request by hairyhenderson**
**What this PR does / why we need it**:

Fixes `warning: cannot overwrite table with non table for custom (map[])`

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #7585 

**Special notes for your reviewer**:

Signed-off-by: Dave Henderson <dhenderson@gmail.com>

### CODE REVIEW

**Comment by steven-sheehy** [comment]
Good catch. Can you also change the README?

File: stable/prometheus-adapter/values.yaml [hunk 1]
```diff
@@ -40,7 +40,7 @@ resources:
 
 rules:
   default: true
-  custom: {}
+  custom: []
```
Comment (davidkarlsen) [inline stable/prometheus-adapter/values.yaml:43]: Please update the readme to match too. TIA

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

File: stable/prometheus-adapter/values.yaml (hunk 1 above)
Comment (hairyhenderson) [reply stable/prometheus-adapter/values.yaml:43]: Thanks, done!

**Review by davidkarlsen — APPROVED** [review]

**Pull request merged**